### PR TITLE
[build] Create Microsoft.iOS.Windows.Sdk workload pack

### PR DESCRIPTION
**Pull request by pjcollins**
Converts the Microsoft.iOS.Windows.Sdk NuGet package into a proper
[workload SDK pack][0].  The entry point for this pack has been changed,
and it is now imported through the `WorkloadManifest.targets` file
included in `Microsoft.NET.Workload.iOS`, rather than being imported
directly from `Microsoft.iOS.Sdk`.

Import ordering has otherwise changed slightly.  The following files are
now imported before the majority of the `Microsoft.iOS.Sdk` (and the 
majority of the .NET SDK targets):

 * Xamarin.iOS.Common.Before.props
 * Xamarin.iOS.Common.Before.targets

After this the majority of the .NET SDK targets will load, followed by
the `Microsoft.iOS.Sdk` targets. Finally, everything declared in the
`<AfterMicrosoftNETSdkTargets/>` hook loads, which consists of:

 * Microsoft.iOS.Windows.Sdk.targets
 * tools/msbuild/*

[0]: https://github.com/dotnet/designs/blob/main/accepted/2020/workloads/workload-manifest.md#sdk-packs

### CODE REVIEW

File: dotnet/Makefile
Comment (rolfbjarne) [inline dotnet/Makefile:109]: This way the file will be updated when the source changes (and also you won't have to change the line below):

```suggestion
Microsoft.NET.Workload.$(1)/WorkloadManifest.json: $4 Makefile $(TOP)/Make.config.inc $(TOP)/.git/HEAD $(TOP)/.git/index
```

File: dotnet/targets/Xamarin.Shared.Sdk.props
Comment (rolfbjarne) [inline dotnet/targets/Xamarin.Shared.Sdk.props:13]: Renaming `_UsingXamarinSdk` to `UsingAppleNETSdk` can be split out into a separate PR.

File: dotnet/Makefile
Comment (rolfbjarne) [inline dotnet/Makefile:125]: I think this will stop generating workloads for platforms included in DOTNET_WINDOWS_PLATFORMS, are you sure you don't want to add a line instead:

```suggestion
$(foreach platform,$(DOTNET_PLATFORMS),$(eval $(call WorkloadTargets,$(platform),$(shell echo $(platform) | tr A-Z a-z),$($(platform)_NUGET_VERSION_NO_METADATA,targets/WorkloadManifest.template.json))))
$(foreach platform,$(DOTNET_WINDOWS_PLATFORMS),$(eval $(call WorkloadTargets,$(platform),$(shell echo $(platform) | tr A-Z a-z),$($(platform)_NUGET_VERSION_NO_METADATA,targets/WorkloadManifest.windows.template.json))))
```

File: dotnet/targets/Xamarin.Shared.Sdk.props
Comment (pjcollins) [reply dotnet/targets/Xamarin.Shared.Sdk.props:13]: Do we want one of these per platform/SDK type, or is `UsingAppleNETSdk` preferred? Android currently sets `UsingAndroidNETSdk`.  I suppose something more granular could also be added in the future if needed.

Comment (pjcollins) [reply dotnet/targets/Xamarin.Shared.Sdk.props:13]: Moved to https://github.com/xamarin/xamarin-macios/pull/11270.

File: dotnet/Makefile
Comment (pjcollins) [reply dotnet/Makefile:125]: I want to use the new "windows" template file for all workloads that include a windows SDK.  These changes _should_ result in the `Microsoft.NET.Workload.iOS` workload declaring this new Windows only pack, however since it's conditionally imported in `WorkloadManifest.targets` it will never need to be installed on macOS.  This all seemed to work locally but it looks like CI didn't run for this yesterday so I'm still waiting to check an official build.  I don't want to introduce new Windows specific workloads (just one new workload pack).  We should be able to continue to ship the same `Microsoft.NET.Workload.iOS` workload on both macOS and Windows.

**Review by rolfbjarne — APPROVED** [review]

File: dotnet/targets/WorkloadManifest.windows.template.json
Comment (pjcollins) [reply dotnet/targets/WorkloadManifest.windows.template.json:3]: The reasoning behind the two files is that `WorkloadManifest.template.json` is used for all other platforms (mac, watch, tv), and these workloads don't (and some will never?) have a `*.Windows.Sdk` entry.

Rather than adding more complex string manipulation to `dotnet/Makefile` to conditionally add this windows entry to _only_ the iOS manifest, I added a new template for workloads that should include a "Windows" Sdk.

We may want add the `"platforms": [ "win-x86", "win-x64" ]` declaration here, though the conditional import in the iOS `WorkloadManifest.targets` file allows the new manifest which declares a windows specific pack to work on macOS when the pack is not installed.

Comment (jonathanpeppers) [reply dotnet/targets/WorkloadManifest.windows.template.json:3]: Ok makes sense 👍 

I think we need to put `platforms` so that the future `dotnet workload install` commands won't install these Windows packs on Mac.

Comment (pjcollins) [reply dotnet/targets/WorkloadManifest.windows.template.json:3]: It looks like this feature may not be working yet, I'm going to revert this portion for now https://github.com/dotnet/sdk/issues/17151

**Review by jonathanpeppers — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Waiting on review from @emaf to merge.

**Review by emaf — APPROVED** [review]

**Pull request merged**
